Sources/Vexil/Flag.swift

```diff
@@ -28,17 +28,92 @@ import Foundation
 @propertyWrapper
 public struct Flag<Value>: Decorated, Identifiable where Value: FlagValue {
 
+    // FlagContainers may have many flags, so to reduce code bloat
+    // it's important that each Flag have as few stored properties
+    // (with nontrivial copy behavior) as possible. We therefore use
+    // a single `Allocation` for all of Flag's stored properties.
+    final class Allocation {
```

Comment:
Can we please move this to the bottom of the file or into a separate file? Having to wade through related types to get to the implementation of the type whose file I opened reduces readability imo. This feedback also applies to `FlagGroup`